### PR TITLE
Add support for isorec flashing in recovery installer

### PR DESCRIPTION
**Pull request by ata2001**

### CODE REVIEW

**Comment by ata2001** [comment]
Please test if you have a device with isorec (only at your own risk, read the diff beforehand)!

**Comment by ollieparanoid** [comment]
@drebrez: I did not make a code review yet (will try to do that tomorrow), so I can't say what's my opinion on that. But in general I'm highly in favor of splitting code up in small files and functions where it makes sense.

File: pmb/install/recovery.py [hunk 1]
```diff
@@ -42,8 +45,17 @@ def create_zip(args, suffix):
               "w") as install_options:
         install_options.write(
             "\n".join(['DEVICE="{}"'.format(args.device),
-                       'FLASH_BOOTIMG="{}"'.format(
-                           str(args.recovery_flash_bootimg).lower()),
+                       'FLAVOR="{}"'.format(flavor),
```
Comment (ollieparanoid) [inline pmb/install/recovery.py]: Proposal to make this part a bit more readable (not tested, probably PEP8 errors!):

```patch
--- a/pmb/install/recovery.py
+++ b/pmb/install/recovery.py
@@ -41,31 +41,30 @@ def create_zip(args, suffix):
     logging.info("(" + suffix + ") create recovery zip")

     # Create config file for the recovery installer
-    with open(args.work + "/chroot_" + suffix + "/tmp/install_options",
-              "w") as install_options:
-        install_options.write(
-            "\n".join(['DEVICE="{}"'.format(args.device),
-                       'FLAVOR="{}"'.format(flavor),
-                       'FLASH_KERNEL="{}"'.format(
-                           str(args.recovery_flash_kernel).lower()),
-                       'ISOREC="{}"'.format(str(method == "heimdall-isorec")
-                                            .lower()),
-                       'KERNEL_PARTLABEL="{}"'.format(
-                           vars["$PARTITION_KERNEL"]),
-                       'INITFS_PARTLABEL="{}"'.format(
-                           vars["$PARTITION_INITFS"]),
-                       'SYSTEM_PARTLABEL="{}"'.format(
-                           vars["$PARTITION_SYSTEM"]),
-                       'INSTALL_PARTITION="{}"'.format(
-                           args.recovery_install_partition),
-                       'CIPHER="{}"'.format(args.cipher),
-                       'FDE="{}"'.format(
-                           str(args.full_disk_encryption).lower())]))
+    options = {
+        "DEVICE": args.device,
+        "FLAVOR": flavor,
+        "FLASH_KERNEL": args.recovery_flash_kernel,
+        "ISOREC": method == "heimdall-isorec",
+        "KERNEL_PARTLABEL": vars["$PARTITION_KERNEL"],
+        "INITFS_PARTLABEL": vars["$PARTITION_INITFS"],
+        "SYSTEM_PARTLABEL": vars["$PARTITION_SYSTEM"],
+        "INSTALL_PARTITION": args.recovery_install_partition,
+        "CIPHER": args.cipher,
+        "FDE": args.full_disk_encryption,
+    }

+    # Write to a temporary file
+    config_temp = args.work + "/chroot_" + suffix + "/tmp/install_options"
+    with open(config_temp, "w") as handle:
+        for key, value in options.keys():
+            if isinstance(value, bool):
+                value = str(value).lower()
+            handle.write(key + "='" + value + "'\n")
      commands = [
```

**Comment by ata2001** [comment]
@ollieparanoid Applied your patch, only has to change options.keys() to options.items().

**Comment by ata2001** [comment]
@ollieparanoid added license header

**Pull request merged**